### PR TITLE
Separate features into persistent and session scoped cookies

### LINKED ISSUE #2932: Update `fetch_sensitive` flag to use session cookie instead of sessionStorage
## Problem

The problem is described in the comments https://github.com/WordPress/openverse/pull/2880#pullrequestreview-1603945685. 
I opened this issue to merge #2880, and will update this description later.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #2932 by @obulat

## Description

This PR
- removes usage of `sessionStorage` from the feature flag store
- migrates session-scoped preferences to the `sessionFeatures` cookie
- makes the `features` cookie persistent (which it wasn't earlier!)
- removes unused code from store and layouts

## Testing Instructions

1. Check out the PR and run the dev server.
2. Toggle some persistent preferences in `/preferences`. E.g.
   - `fake_sensitive`
   - `additional_search_types`
   - `sensitive_content`
4. Restart the browser to see that the changes are preserved.
5. Toggle some session-scoped preferences in `/preferences`. E.g.
   - `fetch_sensitive` (there's only one).
6. Restart the browser to see that the changes are not preserved.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
The failing tests have to be flaky.
image

File: frontend/src/middleware/middleware.ts [hunk 1]
```diff
@@ -32,15 +32,18 @@ const middleware: Middleware = async ({
   /* Feature flag store */
 
   const featureFlagStore = useFeatureFlagStore($pinia)
-  featureFlagStore.initFromCookies($cookies.get("features") ?? {})
+  featureFlagStore.initFromCookies([
```
Comment (obulat) [inline frontend/src/middleware/middleware.ts:35]: In the future, we should move some of the functionality here to a plugin that runs on the first app run. `middleware` is run before every navigation, so we are actually initializing stores from cookies before every page change.

File: frontend/src/stores/feature-flag.ts [hunk 2]
```diff
@@ -122,34 +121,13 @@ export const useFeatureFlagStore = defineStore(FEATURE_FLAG, {
...
-      const features = useStorage<Record<string, FeatureState>>(
-        "features",
-        {},
-        sessionStorage
-      )
-      Object.entries(this.flags).forEach(([name, flag]) => {
-        if (getFlagStatus(flag) === SWITCHABLE && flag.storage === SESSION) {
-          Vue.set(flag, "preferredState", features.value[name])
+        if (getFlagStatus(flag) === SWITCHABLE) {
+          const cookie = cookies.find((cookie) =>
```
Comment (obulat) [inline frontend/src/stores/feature-flag.ts:127]: Can the finding of the cookie value be simplified here? I don't fully understand what's going on.

**Review by obulat — COMMENTED** [review]
Nice start! I'll test it locally when the PR is ready for review.

File: frontend/src/middleware/middleware.ts (hunk 1 above)
Comment (dhruvkb) [reply frontend/src/middleware/middleware.ts:35]: Fair point, we should make an issue to address it separately.

**Review by sarayourfriend — APPROVED** [review]
LGTM! I found https://github.com/WordPress/openverse/issues/2994 while testing this, but it's on `main` so not a problem with this PR. As you can tell, I did some weird edge case testing and this all works correctly, as much as I tried to find an edge case that would break it :sweat_smile:

File: frontend/src/layouts/content-layout.vue [hunk 3]
```diff
@@ -63,11 +62,6 @@ export default defineComponent({
   setup() {
     const uiStore = useUiStore()
 
-    const featureStore = useFeatureFlagStore()
```
Comment (obulat) [inline frontend/src/layouts/content-layout.vue:66]: So nice that we can remove this code from the layouts!

**Review by obulat — APPROVED** [review]
Awesome work 🎉  Everything works great locally

**Comment by dhruvkb** [comment]
The code where Sentry says the error originated was _removed by this PR_!

**Comment by obulat** [comment]
> The code where Sentry says the error originated was removed by this PR!
I think this is caused by this PR being deployed to the staging (so, Sentry thinks this PR was deployed), but not released in prod (so, the issue keeps occurring).

**Comment by dhruvkb** [comment]
Ah, that makes sense, thanks @obulat! But in that case, Sentry should not reporting line numbers relative to the current commit and not `main`.

**Comment by obulat** [comment]
The releases might not be set up correctly in Sentry, @dhruvkb .

**Pull request merged**